cpp/src/scalar/scalar.cpp

```diff
@@ -56,4 +95,352 @@ std::string string_scalar::to_string(rmm::cuda_stream_view stream) const
+                                                          rmm::cuda_stream_view stream,
+                                                          rmm::mr::device_memory_resource* mr);
+template timestamp_scalar<timestamp_us>::timestamp_scalar(duration_us const& value,
+                                                          bool is_valid,
+                                                          rmm::cuda_stream_view stream,
+                                                          rmm::mr::device_memory_resource* mr);
+template timestamp_scalar<timestamp_us>::timestamp_scalar(int64_t const& value,
+                                                          bool is_valid,
+                                                          rmm::cuda_stream_view stream,
+                                                          rmm::mr::device_memory_resource* mr);
+template timestamp_scalar<timestamp_ns>::timestamp_scalar(duration_D const& value,
+                                                          bool is_valid,
+                                                          rmm::cuda_stream_view stream,
+                                                          rmm::mr::device_memory_resource* mr);
+template timestamp_scalar<timestamp_ns>::timestamp_scalar(duration_s const& value,
+                                                          bool is_valid,
+                                                          rmm::cuda_stream_view stream,
+                                                          rmm::mr::device_memory_resource* mr);
+template timestamp_scalar<timestamp_ns>::timestamp_scalar(duration_ms const& value,
+                                                          bool is_valid,
+                                                          rmm::cuda_stream_view stream,
+                                                          rmm::mr::device_memory_resource* mr);
+template timestamp_scalar<timestamp_ns>::timestamp_scalar(duration_us const& value,
+                                                          bool is_valid,
+                                                          rmm::cuda_stream_view stream,
+                                                          rmm::mr::device_memory_resource* mr);
+template timestamp_scalar<timestamp_ns>::timestamp_scalar(duration_ns const& value,
+                                                          bool is_valid,
+                                                          rmm::cuda_stream_view stream,
+                                                          rmm::mr::device_memory_resource* mr);
+template timestamp_scalar<timestamp_ns>::timestamp_scalar(int64_t const& value,
+                                                          bool is_valid,
+                                                          rmm::cuda_stream_view stream,
+                                                          rmm::mr::device_memory_resource* mr);
```

Comment:
Such long explicit template instantiation can also be shortened by my recommendation above.

Reply:
I think listing the parameters here make this more difficult to update. Removing the names did not seem to help. 
So I created a macro for this case so only one place needs to change if the parameter or definition changes.